src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -190,7 +191,12 @@ impl LeaderBlockCommitOp {
         let parent_vtxindex = parse_u16_from_be(&data[68..70]).unwrap();
         let key_block_ptr = parse_u32_from_be(&data[70..74]).unwrap();
         let key_vtxindex = parse_u16_from_be(&data[74..76]).unwrap();
-        let burn_parent_modulus = ((data[76] as u64) % BURN_BLOCK_MINED_AT_MODULUS) as u8;
+
+        let burn_parent_modulus_and_memo_byte = data[76];
+
+        let burn_parent_modulus = ((burn_parent_modulus_and_memo_byte & 0b111) as u64
+            % BURN_BLOCK_MINED_AT_MODULUS) as u8;
+        let memo = (burn_parent_modulus_and_memo_byte >> 3) & 0x1f;
```

Comment:
The `& 0x1f` shouldn't be necessary, strictly speaking, because it's a `u8`.  If you wanted to be explicit about the masking though (which I support), you could do it as `(burn_parent_modulus_and_memo_byte & 0b11111000) >> 3`.

Reply:
Yeah -- I don't think it's necessary, it's added here because the first source I found for rust's shift right operand said that it would shift in the high bit (i.e., `0b1111 >> 1 = 0b1111` rather than `0b0111`), but that's not the case -- right shift on uints shifts in `0` (`(burn_parent_modulus_and_memo_byte & 0b11111000) >> 3` wouldn't insure that 0 bits were shifted in).